addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -313,3 +401,35 @@ func (r *MirrorPeerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&multiclusterv1alpha1.MirrorPeer{}, builder.WithPredicates(mpPredicate)).
 		Complete(r)
 }
+
+func (r *MirrorPeerReconciler) labelRBDStorageClasses(ctx context.Context, mp multiclusterv1alpha1.MirrorPeer, storageClusterNamespace string) []error {
+	// Get all StorageClasses in storageClusterNamespace
+	scs := &storagev1.StorageClassList{}
+	klog.Infof("Getting all StorageClasses in namespace %q", storageClusterNamespace)
+	err := r.SpokeClient.List(ctx, scs, client.InNamespace(storageClusterNamespace))
+	var errs []error
+	if err != nil {
+		errs = append(errs, err)
+		return errs
+	}
+	klog.Infof("Found %d StorageClasses in namespace %q", len(scs.Items), storageClusterNamespace)
+	key := r.SpokeClusterName
+	for _, sc := range scs.Items {
+		if _, ok := clusterFSIDs[key]; !ok {
+			errs = append(errs, fmt.Errorf("No value found for key: %s, unable to update StorageClass for %s", key, key))
+			continue
+		}
+		if sc.Provisioner == fmt.Sprintf(RBDProvisionerTemplate, storageClusterNamespace) {
```

Comment:
can't we filter this while getting the storageclass list? I think there is a filter option.

Reply:
tried filtering it using field selectors but we also need to label cephfs storage classes. Field selectors allow only AND based conditions not OR based